lib/web/css/source/lib/variables/_forms.less

```diff
@@ -49,7 +49,7 @@
 //  Form elements choice default variables
 @form-element-choice__type: ''; // [radio|checkbox]
 @form-element-choice__vertical-align: false;
-@form-element-choice__margin: 2px @indent__xs 0 0;
+@form-element-choice__margin: -2px @indent__xs 0 0;
```

Comment:
It's also changing in general styles for all checkboxes in Magento, but we have the issue only in one specific case. I think you should fix styles only for the current case because on other pages we haven't an alignment issue.